src/js/overlay.js

```diff
@@ -243,10 +253,15 @@ Overlay.prototype.show = function() {
 	if (!this.opts.nested) {
 		this.resizeListenerHandler = this.resizeListener.bind(this);
 		this.delegates.doc.on('oViewport.resize', 'body', this.resizeListenerHandler);
-	}
 
-	this.closeOnNewLayerHandler = this.closeOnNewLayer.bind(this);
-	this.delegates.context.on('oLayers.new', this.closeOnNewLayerHandler);
+		this.closeOnNewLayerHandler = this.closeOnNewLayer.bind(this);
+		this.delegates.context.on('oLayers.new', this.closeOnNewLayerHandler);
+
+		this.broadcast('new', 'oLayers');
```

Comment:
@rowanbeentje Should we still fire `this.broadcast('new', 'oLayers');` in nested cases? I think the problem is solved by moving the listener `closeOnNewLayerHandler`. I'm still getting to grips with `o-overlay` and `o-layer` and could be wrong. 😅

Reply:
Open to debate I think :D  My thinking was that layers is for tracking things which use the z-axis to overlay content, and in general with these notifications it's so that overlays don't end up stacking on the screen.  Nested overlays are more like a call-to-action somewhere on the page (eg the feedback example on the registry), so aren't really creating a new layer - they're just a styled component in the page rather than something stacked on top of it...

Reply:
After discussion: yes, the current `position: relative` for nested demos and usage is implied and added by external Sass, so by default it is added as a new layer, so *firing* the event still makes sense...